docs/encryption.md

```diff
@@ -0,0 +1,245 @@
+# Data-at-rest encryption
+
+Data-at-rest encryption ensures that stored data remains protected even if the underlying storage is compromised. Percona Operator for MySQL uses the `keyring_vault` plugin, shipped with Percona Server for MySQL, to encrypt tablespaces, bin logs and backups. 
+
+[HashiCorp Vault :octicons-link-external-16:](https://www.vaultproject.io/) is used to securely store and manage encryption keys, enabling automatic key rotation, audit logging, and compliance with enterprise security standards. This setup enhances the overall security posture of your MySQL cluster.
+
+This guide walks you through deploying and configuring HashiCorp Vault to work with Percona Operator for MySQL to enable data-at-rest encryption. 
+## Create the namespace
+
+It is a good practice to isolate workloads in Kubernetes using namespaces. Create a namespace with the following command:
+
+```{.bash .data-prompt="$"}
+$ kubectl create namespace vault
+```
+
+Export the namespace as an environment variable to simplify further configuration and management
+
+```bash 
+NAMESPACE="vault"
+```
+
+## Install Vault
+
+For this setup, we install Vault in Kubernetes using the [Helm 3 package manager :octicons-link-external-16:](https://helm.sh/). However, Helm is not required — any supported Vault deployment (on-premises, in the cloud, or a managed Vault service) works as long as the Operator can reach it.
+
+1. Add and update the Vault Helm repository.
+
+    ``` {.bash data-prompt="$" }
+    $ helm repo add hashicorp https://helm.releases.hashicorp.com
+    $ helm repo update
+    ```
+
+2. Install Vault 
+
+    ``` {.bash data-prompt="$" }
+    $ helm upgrade --install vault hashicorp/vault \ 
+      --namespace $NAMESPACE \
+    ```
+
+    ??? example "Sample output"
+
+        ```{.text .no-copy}
+        NAME: vault2
```

Comment:
It's a bit strange that the release name during installation is vault but the name here is vault 2. And the rest of the doc uses vault2. Should it be just vault?